app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -365,10 +385,61 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
 
     private fun updateAccountProfile(profile: Profile) {
         launch {
-            val preferenceFirefoxAccount =
-                findPreference<Preference>(context!!.getPreferenceKey(pref_key_account))
-            preferenceFirefoxAccount?.title = profile.displayName.orEmpty()
-            preferenceFirefoxAccount?.summary = profile.email.orEmpty()
+            context?.let { context ->
+                val preferenceFirefoxAccount =
+                    findPreference<AccountPreference>(context.getPreferenceKey(pref_key_account))
+
+                preferenceFirefoxAccount?.title?.setTextColor(
+                    R.attr.primaryText.getColorFromAttr(context)
+                )
+                preferenceFirefoxAccount?.title?.text = profile.displayName.orEmpty()
+                preferenceFirefoxAccount?.title?.visibility =
+                    if (preferenceFirefoxAccount?.title?.text.isNullOrEmpty()) View.GONE else View.VISIBLE
+
+                preferenceFirefoxAccount?.summary?.setTextColor(
+                    R.attr.primaryText.getColorFromAttr(context)
+                )
+                preferenceFirefoxAccount?.summary?.text = profile.email.orEmpty()
+
+                preferenceFirefoxAccount?.icon = ContextCompat.getDrawable(context, R.drawable.ic_shortcuts)
+                preferenceFirefoxAccount?.errorIcon?.visibility = View.GONE
+                preferenceFirefoxAccount?.background?.background = null
+            }
+        }
+    }
+
+    private fun displayAccountErrorIfNecessary() {
+        launch {
+            context?.let { context ->
+                if (!requireComponents.backgroundServices.accountManager.accountNeedsReauth()) { return@launch }
```

Comment:
nit, but let's clean this up before landing:
re-written, what you're doing here is essentially:
```
if (context != null) {
 if (context == null) { throw new IllegalStateException("Fragment " + this + " not attached to a context."}
 // do stuff
}
```
s/`requireComponents`/`context.components`